sdk/core/core-client-rest/test/multipart.spec.ts

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { describe, it, assert } from "vitest";
+import { PartDescriptor, buildBodyPart } from "../src/multipart.js";
+import { stringToUint8Array } from "../../core-util/dist/commonjs/bytesEncoding.js";
+
+describe("multipart buildBodyPart", () => {
+  describe("content-type calculation", () => {
+    it.each([
+      {
+        description: "content type header set to null results in header being unset",
+        descriptor: {
+          contentType: null,
+          body: "even though the body is a string, the content type should still not be set",
+        },
+        expected: undefined,
+      },
+      {
+        description: "content-type header set in part headers takes precedence over contentType",
+        descriptor: {
+          contentType: "dont-pick-me",
+          headers: {
+            "content-type": "pick-me-instead",
+          },
+        },
+        expected: "pick-me-instead",
+      },
+      {
+        description: "contentType value takes precedence over natural content type",
+        descriptor: {
+          contentType: "application/json",
+          body: `{ "aaa": "bbb" }`, // would otherwise expect this to give text/plain; charset=UTF-8
+        },
+        expected: "application/json",
+      },
+      {
+        description: "content type is taken from Blob if available",
```

Comment:
a UT for type is customized in File?

Reply:
I don't think this is necessary since `File` extends `Blob`.